src/components/application_manager/src/commands/mobile/subscribe_vehicle_data_request.cc

```diff
@@ -429,5 +335,120 @@ bool SubscribeVehicleDataRequest::IsSomeoneSubscribedFor(
+      if (!is_interface_not_available && is_key_enabled) {
+        VehicleDataType key_type = it->second;
+        if (app->IsSubscribedToIVI(key_type)) {
+          LOG4CXX_DEBUG(logger_,
+                        "App with connection key "
+                            << connection_key()
+                            << " is subscribed already for VehicleDataType: "
+                            << key_type);
+          ++subscribed_items;
+          vi_already_subscribed_by_this_app_.insert(key_type);
+          out_response_params[key_name][strings::data_type] = key_type;
+          out_response_params[key_name][strings::result_code] =
+              mobile_apis::VehicleDataResultCode::VDRC_DATA_ALREADY_SUBSCRIBED;
+          continue;
+        }
+
+        if (IsSomeoneSubscribedFor(key_type)) {
+          LOG4CXX_DEBUG(logger_,
+                        "There are apps subscribed already for "
+                        "VehicleDataType: "
+                            << key_type);
+          if (!app->SubscribeToIVI(static_cast<uint32_t>(key_type))) {
+            LOG4CXX_ERROR(
+                logger_,
+                "Unable to subscribe for VehicleDataType: " << key_type);
+            continue;
+          }
+          LOG4CXX_DEBUG(
+              logger_,
+              "App with connection key "
+                  << connection_key()
+                  << " have been subscribed for VehicleDataType: " << key_type);
+          ++subscribed_items;
+          vi_already_subscribed_by_another_apps_.insert(key_type);
+          out_response_params[key_name][strings::data_type] = key_type;
+          out_response_params[key_name][strings::result_code] =
+              mobile_apis::VehicleDataResultCode::VDRC_SUCCESS;
+          continue;
+        }
+
+        out_request_params[key_name] = is_key_enabled;
+
+        if (app->SubscribeToIVI(static_cast<uint32_t>(key_type))) {
+          LOG4CXX_DEBUG(
+              logger_,
+              "App with connection key "
+                  << connection_key()
+                  << " have been subscribed for VehicleDataType: " << key_type);
+          ++subscribed_items;
+        }
+      }
+    }
+  }
+
+  const bool is_everything_already_subscribed =
+      static_cast<uint32_t>(items_to_subscribe) ==
+      vi_already_subscribed_by_another_apps_.size() +
+          vi_already_subscribed_by_this_app_.size();
+
+  if (0 == items_to_subscribe) {
+    if (HasDisallowedParams()) {
+      out_result_code = mobile_apis::Result::DISALLOWED;
+    } else {
+      out_result_code = mobile_apis::Result::INVALID_DATA;
+      out_info = "No data in the request";
+    }
+    result = false;
+  }
+
+  if (0 == subscribed_items && !is_interface_not_available) {
+    out_result_code = mobile_apis::Result::IGNORED;
+    out_info = "Already subscribed on provided VehicleData.";
+    result = false;
+  }
+
+  if (is_everything_already_subscribed) {
+    out_result_code = vi_already_subscribed_by_this_app_.size()
```

Comment:
@dtrunov This comparison is always `true`. Did you mean `==` ? Unfortunately it results in signed versus unsigned comparison. This will fail to compile under Linux. Also it is confusing to compare `enum` to container size.